.github/workflows/e2e-k8s.yml

```diff
@@ -0,0 +1,36 @@
+name: e2e-on-k8s
```

Comment:
why not just `e2e`?

Reply:
we have another workflow `e2e-without-lifecycle-manager`, which should be renamed to `e2e` IMO. Thats why kept the k8s to convey that this workflow uses real k8s on gardener.

Reply:
How about we use the following prefixes when it make sense `e2e-k3e` `e2e-gardener`?

Reply:
I will do it in next PR.